### PR TITLE
Upgrade github.com/tinkerbell/{tink,rufio} dependencies:

### PR DESCRIPTION
**Pull request by jacobweinstock**
*Issue #, if available:*

*Description of changes:*
This is needed for VLAN ID support as the latest upstream CAPT uses these same versions.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/diagnostics/collectors.go [hunk 1]
```diff
@@ -317,9 +317,9 @@ func (c *collectorFactory) snowCrdCollectors() []*Collect {
 
 func (c *collectorFactory) tinkerbellCrdCollectors() []*Collect {
 	captCrds := []string{
-		"baseboardmanagements.bmc.tinkerbell.org",
-		"bmcjobs.bmc.tinkerbell.org",
-		"bmctasks.bmc.tinkerbell.org",
+		"machines.bmc.tinkerbell.org",
```
Comment (abhinavmpandey08) [inline pkg/diagnostics/collectors.go:320]: Did we rename this CRD to `machines` upstream?
CAPI already has a CRD named `machines` so this is gonna be ambiguous

**Review by abhinavmpandey08 — COMMENTED** [review]
Will this break compatibility with older EKS-A versions since we are re-naming the CRDs?

Also, is build-tooling updated to accommodate these changes?

File: pkg/diagnostics/collectors.go (hunk 1 above)
Comment (jacobweinstock) [reply pkg/diagnostics/collectors.go:320]: we did. https://github.com/tinkerbell/rufio/blob/main/config/crd/bases/bmc.tinkerbell.org_machines.yaml
It's namespaced though.

Comment (ptrivedi) [reply pkg/diagnostics/collectors.go:320]: I agree that a CRD named `machines` is going to be confusing and ambiguous

Comment (chrisdoherty4) [reply pkg/diagnostics/collectors.go:320]: Custom resources are namespaced so there aren't any collisions from a technical point of view; I can appreciate humans may do a double take. There are numerous projects that define the same kind across different groups (namespaces). For example, [argo-cd](https://doc.crds.dev/github.com/argoproj/argo-cd) and [litmus](https://doc.crds.dev/github.com/litmuschaos/litmus) define `Workflow` and `CronWorkflow` types. [Crossplane](https://doc.crds.dev/github.com/crossplane/crossplane) defines 2 `Configuration` resources in the same project but under different namespaces. There are many other examples.

Its infeasible, and defeats the point of namespacing, to expect projects to need to create unique names - not to mention this is upstream so can't be changed here.

**Comment by chrisdoherty4** [comment]
Completed as a separate piece of work.

**Pull request closed without merging**